Dockerfile

```diff
@@ -0,0 +1,32 @@
+# stage1 as builder
+FROM node:alpine as builder
+
+# copy the package.json to install dependencies
+COPY package.json package-lock.json ./
+
+# Install the dependencies and make the folder
+RUN npm install && mkdir /react-ui && mv ./node_modules ./react-ui
+
+WORKDIR /react-ui
+
+COPY . .
+
+# Build the project and copy the files
+RUN npm run build
+
+
+FROM nginx:alpine
+
+#!/bin/sh
+
+COPY ./.nginx/nginx.conf /etc/nginx/nginx.conf
+
+## Remove default nginx index page
+RUN rm -rf /usr/share/nginx/html/*
+
+# Copy from the stahg 1
+COPY --from=builder /react-ui/build /usr/share/nginx/html
+
+EXPOSE 3000 3000
```

Comment:
From my understanding, `EXPOSE <Port Number>` is used to make a service or function available to other Docker containers, but not outside Docker. Where exactly is this feature utilized and how?

Reply:
To be able to globally expose the port outside of the container environment we have a `ports` section in the `docker-compose.yml` file which sets `3000:3000`. The docker-compose is what is responsible for taking an image and defining a runtime environment for it. By having that port section we globally expose that port so that the host machine can connect within the container over port 3000. The EXPOSE in the Dockerfile ensures that there's something within the container listening at the container's port 3000 and maps that over to our React application.